src/prop/cadical.cpp

```diff
@@ -1312,6 +1539,35 @@ std::vector<Node> CadicalSolver::getOrderHeap() const { return {}; }
 
 std::shared_ptr<ProofNode> CadicalSolver::getProof()
 {
+  if (d_proof_tracer)
+  {
+    ProofNodeManager* pnm = d_env.getProofNodeManager();
+    NodeManager* nm = d_env.getNodeManager();
+
+    std::vector<SatClause> unsat_core;
+    d_proof_tracer->compute_unsat_core(unsat_core);
+
+    std::vector<std::shared_ptr<ProofNode>> ps;
+    for (const auto& sat_clause : unsat_core)
+    {
+      NodeBuilder nb(nm, Kind::OR);
+      std::vector<Node> lits;
+      for (const auto& lit : sat_clause)
+      {
+        lits.push_back(d_proxy->getNode(lit));
+      }
+      // Sat clause is sorted by literal id. Ensure that node-level clause is
+      // sorted by node ids.
+      std::sort(lits.begin(), lits.end());
+      for (const auto& lit : lits)
+      {
+        nb << lit;
+      }
+      Node n = nb.getNumChildren() == 1 ? nb[0] : nb.constructNode();
+      ps.push_back(pnm->mkAssume(n));
+    }
+    return pnm->mkNode(ProofRule::SAT_REFUTATION, ps, {});
```

Comment:
Should we only return a SAT_REFUTATION here if the `d_proof_tracer` is present?  Otherwise if it is off then ps is empty, which would be wrong.

Reply:
This is what this block does. There is a return null further down below for the case where we don't have a proof tracer.

Reply:
Ah sorry, misread the indentation.